### PR TITLE
[LLVM update] Fix CI segfaults in the xfailed grad+vmap tests

### PR DESCRIPTION
**Pull request by mehrdad2m**
**Context:**
There has been a old issue with using grad + vmap captured by https://github.com/PennyLaneAI/catalyst/issues/1912 that seems to be partially resolved as we updated LLVM + Enzyme. This most likely related to the update from enzyme. As a result of this three of the tests previously marked  as xfail now have changed status. One passes and the other two segfault (yikes!). Turns out that the example discussed in https://github.com/PennyLaneAI/catalyst/issues/1912 which is the floowing:
```python
@vmap
@qml.qnode(qml.device("lightning.qubit", wires=1))
def circuit(x):
    qml.RX(x, wires=0)
    return qml.expval(qml.PauliZ(0))

@qjit
@grad
def loss(x):
    return jnp.sum(circuit(x))

print(loss(jnp.array([0.1, 0.2, 0.3])))
```
now runs with no compilation error. However a case like below segfaults at runtime:

```python 
@vmap
@qml.qnode(qml.device("lightning.qubit", wires=1))
def circuit(x, data):
    qml.RX(x, wires=0)
    qml.RX(data, wires=0)
    return qml.expval(qml.PauliZ(0))

@qjit
@grad
def loss(x, data):
    return jnp.sum(circuit(x, data))

print(loss(jnp.array([0.1, 0.2, 0.3]), jnp.array([0.1, 0.2, 0.3])))
```
the main difference is the addition of `data` as the second argument.

**Description of the Change:**
Remove the xfail for the passed test and move the other two xfails inside the test to avoid their execution and hence the segmentation fault.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by paul0403 — APPROVED** [review]

**Comment by dime10** [comment]
Looking at the tests in question, it's weird that the error there could have gone away by updating Enzyme, because they reference PR #332, which fixes a bug in our gradient transform wrt some instruction placement. At the time we decided not to merge 332 because we figured a compiler error was better than a crash (that arose once we fixed the compiler error). If the error is no longer raised anyway, I think it would be worth reviving 332 to see if the issue described there has somehow gone away or else merge it.

**Review by kipawaa — APPROVED** [review]
